src/mumble/dialogs/PositionalAudioViewer.h

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_DIALOGS_POSITIONALAUDIOVIEWER_H_
+#define MUMBLE_MUMBLE_DIALOGS_POSITIONALAUDIOVIEWER_H_
+
+#include "PositionalData.h"
+
+#include <memory>
+
+#include <QDialog>
+#include <QTimer>
+
+namespace Ui {
+class PositionalAudioViewer;
+};
+
+class PositionalAudioViewer : public QDialog {
```

Comment:
Why not let `PositionalAudioViewer` directly extend `Ui::PositionalAudioViewer`? What is this indirection of having the actual GUI class as a member variable good for? :thinking:

Reply:
There are 3 common ways in Qt:

1. Inherit the `Ui` class.
2. Declare a pointer to an `Ui` class object (allocated on the heap).
3. Declare an `Ui` class object (allocated on the stack).

I personally always used the second method, mainly because it lets me include the `ui_` header in the `.cpp` file.

As a result, if you modify the `.ui` file, only a single `.cpp` file is recompiled.

With the first and third method, instead, all files that include the header would get recompiled.

Reply:
Good point :+1: 
I guess we could adapt this style in a bunch of places though since all examples that I found 8and used) in our source code use the first version...

Reply:
Yeah, I actually only recently noticed it.

Qt Creator lets you decide how the classes should be generated, with the second method being the default.